### PR TITLE
React_ui_Increment_Decrement_Filter

### PR DESCRIPTION
**Pull request by saurabhuideveloper**
[story](https://www.pivotaltracker.com/story/show/148394571)

### CODE REVIEW

File: stories/index.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ import {
   ModalNoOverlayClose,
   FilterPanel,
   RangeSlider,
+  Collapse,
+  IncrementDecrementFilter,
+
```
Comment (HansHauge) [inline stories/index.js]: extra new line here

**Comment by HansHauge** [comment]
I'm not actually seeing the prototype, is it just me?

screen shot 2017-07-19 at 10 42 38 am

File: packages/react-ui-core/src/Collapsible/Collapsible.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+export default class Collapsible extends Component {
+  constructor(props) {
+    super(props)
+    this.handleClick = this.handleClick.bind(this)
+    this.state = {
+      display: false,
+      toggleItems: this.props.toggleItems,
+    }
+  }
+  handleClick() {
```
Comment (tcwilson) [inline packages/react-ui-core/src/Collapsible/Collapsible.js]: can you add a newline between methods, to make it more consistent with our current code? See [this file](https://github.com/rentpath/react-ui/blob/master/packages/react-ui-core/src/Form/Field.js) for example.

File: packages/react-ui-core/src/Collapsible/Collapsible.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        <div className={theme.alignBottom}>
+          <a onClick={this.handleClick}>
+            <strong>{toggleItems}</strong>
+          </a>
+        </div>
+    </div>
+    )
+  }
+}
+Collapsible.defaultProps = {
```
Comment (tcwilson) [inline packages/react-ui-core/src/Collapsible/Collapsible.js]: This is an acceptable style when doing functional components, but when creating an object (extending `Component` or `PureComponent`) we like to keep the `propTypes` and `defaultProps` in the object definition, near the top. [A propTypes example](https://github.com/rentpath/react-ui/blob/master/packages/react-ui-core/src/Form/Field.js#L15). [A defaultProps example](https://github.com/rentpath/react-ui/blob/master/packages/react-ui-core/src/Form/Field.js#L109).

File: packages/react-ui-core/src/Collapsible/Collapsible.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      theme,
+      showableItems,
+      nonshowableItems,
+      id,
+      title,
+      checkboxClass,
+    } = this.props
+    const { toggleItems } = this.state
+    return (
+    <div id={id} className={this.props.collapsible}>
```
Comment (tcwilson) [inline packages/react-ui-core/src/Collapsible/Collapsible.js]: indentation here needs to be fixed

**Comment by tcwilson** [comment]
Why is all the Collapsible code in this PR? The ticket is for increment/decrement and there is a separate PR for Collapsible.

Also, we've switched from using mocha to using jest for unit testing. You can do this:
```
cd packages/react-ui-core/
yarn lint
```
To get an easier view of what needs fixed. You can check out [Nino's PR](https://github.com/rentpath/react-ui/pull/26) to see how he converted the tests to using the new jest format.

**Comment by HansHauge** [comment]
I'm seeing some test failures, can someone else confirm / deny?

screen shot 2017-07-24 at 8 34 42 am

File: stories/examples/IncrementDecrementFilter.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+const count = 3
+const IncrementDecrementFilterComponent = (
+  <div>
+    <div className={IncrementDecrementFilterTheme.title}>
+    <p>Baths & Bedrooms</p>
+    </div>
+    <IncrementDecrementFilter theme={IncrementDecrementFilterTheme} rightUnit="+" label="Bedrooms" count={count} />
+    <IncrementDecrementFilter theme={IncrementDecrementFilterTheme} rightUnit="+" label="Bathrooms" count={count} />
+    </div>
+)
```
Comment (tcwilson) [inline stories/examples/IncrementDecrementFilter.js]: can you fix the indentation on this block?

File: stories/theme/IncrementDecrementFilter.css [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+.label {
+  display: table-cell; 
+  vertical-align: middle;
+}
+
+.componentContainer {
+ vertical-align: middle;
+ display: table-cell;
+ width: 400px;
+}
```
Comment (tcwilson) [inline stories/theme/IncrementDecrementFilter.css]: indentation

**Comment by lfender6445** [comment]
please resolve merge conflicts

**Comment by saurabhuideveloper** [comment]
Trying with new PR

**Pull request closed without merging**